src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -440,7 +440,7 @@ protected void collectAll()
 
         // 15-Jan-2023, tatu: [databind#3736] Let's avoid detecting fields of Records
         //   altogether (unless we find a good reason to detect them)
-        if (!isRecordType()) {
+        if (!isRecordType() || _forSerialization) {
```

Comment:
No, I don't think this should be done, even if it handles this particular problem.

Reply:
Ok, thank you @yihtserns -- it's not so much for pulling in getters, but for alternate visibility.

I think this is necessary, then. I am  bit worried about possible side-effects of doing this (wrt forcing access to `Field`s but since it's only for serialization perhaps it's not a problem).

Reply:
I've did test this against that issue, the other day - apparently they're unrelated. 😮‍💨